client/team.go

```diff
@@ -120,8 +134,8 @@ func (c *Client) GetTeam(ctx context.Context, idOrSlug string) (t Team, err erro
 }
 
 type UpdateSamlConfig struct {
-	Enforced bool              `json:"enforced"`
-	Roles    map[string]string `json:"roles"`
+	Enforced bool           `json:"enforced"`
+	Roles    map[string]any `json:"roles"`
```

Comment:
`any` is pretty gnarly here, when we just need to support marshaling, a union type is only a few more lines of code.

Reply:
something like:

```
type UpdatSamlConfigRole struct {
  Role *string
  AccessGroupID *string
}

func (r *UpdateSamlConfigRole) MarshalJSON() ([]byte, error) {
  if r.Role != nil { return json.Marshal(*r.Role) }
  if r.AccessGroupID != nil { return json.Marshal(*r.AccessGroupID) }
  return nil, fmt.Errorf("bad union")
}
```